numpy/core/src/multiarray/array_coercion.c

```diff
@@ -588,7 +592,11 @@ npy_unlink_coercion_cache(coercion_cache_obj *current)
     return next;
 }
 
-NPY_NO_EXPORT NPY_INLINE void
+NPY_NO_EXPORT
+#if (!defined(_MSC_VER) && !defined(__clang__))
+NPY_INLINE
```

Comment:
Does inlining cause a problem?

Reply:
For some reason clang-cl really didn't like this idiom.

Reply:
NPY_INLINE is defined in  `npy_common.h` by
```
#if defined(_MSC_VER)
        #define NPY_INLINE __inline
#elif defined(__GNUC__)
    #if defined(__STRICT_ANSI__)
         #define NPY_INLINE __inline__
    #else
         #define NPY_INLINE inline
    #endif
#else
    #define NPY_INLINE
#endif
```
So it should be blank already unless one of  `_MSC_VER` or `__GNUC__`  is defined.

Reply:
Hmm, I see you have already updated that bit of code.

Reply:
Looks to me like the `NPY_INLINE` macro should just be removed or the function should be defined in `array_coercion.h` that is included in `ctors.c`. The current code in incorrect C.

Reply:
I have removed these here.  I also submitted a PR to remove these irrespective of this PR

Reply:
clang-cl defines `_MSC_VER`, so `NPY_INLINE` is defined by default.

I removed the inline.  I can't imagine there is a detectable perf different from these even if compilers somehow managed to inline them (they probably ignore it).